teps/0072-results-json-serialized-records.md

```diff
@@ -0,0 +1,465 @@
+- [References (optional)](#references-optional)
+  <!-- /toc -->
+
+## Summary / Motivation
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Currently we use the well-known
+[`Any`](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/any.proto) proto
+to represent arbitrary Record data:
+
+```proto
+message Any {
+  string type_url = 1;
+  bytes value = 2;
+}
+```
+
+While this has worked so far, this has some challenges that we would like to address:
+
+1. [Handcrafted proto types are prone to errors](https://github.com/tektoncd/results/issues/101).
+   We ideally need to generate or use the underlying types directly.
+2. [We have had difficulty](https://github.com/tektoncd/results/issues/101#issuecomment-823604704)
+   generating Tekton Pipeline proto types from the underlying Go source due to
+   complications with `knative.dev/pkg` and the Kubernetes protobuf code
+   generator (https://github.com/knative/pkg/issues/2099).
+3. This requires all integrations to generate protobuf types for data they would
+   like to store, which may be a roadblock for integrations if they do not have
+   protobuf definitions already made.
+
+### Goals
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Remove handcrafted Pipeline types (and other project) types in favor of
+  autogenerated/directly defined types.
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Backwards compatibility - the Results API is in alpha, and we intend to lean
+  on this to make breaking changes quickly. We will provide migration tools to
+  help users convert between old formats and new formats.
```

Comment:
Nice :)
I think it would be really beneficial if we were able to keep the data currently collected in the dogfooding cluster.